app/controllers/concerns/listings_toolkit.rb

```diff
@@ -158,13 +137,15 @@ def bump_listing(cost)
 
   def charge_credits_before_bump(purchaser, cost)
     ActiveRecord::Base.transaction do
-      Credits::Buyer.call(
+      enough_credits = Credits::Buy.call(
         purchaser: purchaser,
         purchase: @listing,
         cost: cost,
       )
 
-      raise ActiveRecord::Rollback unless bump_listing_success
+      unless enough_credits && @listing.bump
```

Comment:
I'm not sure why the original code checked for the success of the bumping operation, I don't see how that could go wrong. But I kept it like this for now.